emission/analysis/section_features.py

```diff
@@ -140,14 +162,50 @@ def calVCR(segment):
         else:
             return 0
 
+
+# def calVCR(segment):
+#     trackpoints = segment['track_points']
+#     if len(trackpoints) < 3:
+#         return 0
+#     else:
+#         Pv = 0
+#         for (i, point) in enumerate(trackpoints[:-2]):
+#             currPoint = point
+#             nextPoint = trackpoints[i+1]
+#             nexNextPt = trackpoints[i+2]
+#             velocity1 = calSpeed(currPoint, nextPoint)
+#             velocity2 = calSpeed(nextPoint, nexNextPt)
+#             if velocity1 != None and velocity2 != None:
+#                 if velocity1 != 0:
+#                     VC = abs(velocity2 - velocity1)/velocity1
+#                 else:
+#                     VC = 0
+#             else:
+#                 VC = 0
+
+#             if VC > 0.7:
+#                 Pv += 1
+
+#         segmentDist = segment['distance']
+#         if segmentDist != None and segmentDist != 0:
+#             return Pv/segmentDist
+#         else:
+#             return 0
 def calSegmentDistance(segment):
   return segment['distance']
 
+# def calSpeeds(segment):
+#   trackpoints = (segment['speeds'], segment['distances'])
+#   distances = segment['distances']
+#   if len(trackpoints) == 0:
+#     return None
+#   return calSpeedsForList(trackpoints)
+
 def calSpeeds(segment):
-  trackpoints = segment['track_points']
-  if len(trackpoints) == 0:
-    return None
-  return calSpeedsForList(trackpoints)
+    try:
+        return segment['speeds']
+    except KeyError:
+        return []
```

Comment:
Since in our if statements, we are checking `speeds != None and len(speeds) > 0` it doesn't matter as much here. But in the future, since we're usually expecting a list from speeds, it might be better to return an empty list instead of none, since we might do some list operations, like check the length, later, and if we returned None, it would throw an error. In that way, i think it is safer to return the empty list.